### PR TITLE
Add new SUNDIALS versions

### PR DESCRIPTION
**Pull request by balos1**
Added SUNDIALS 3.1.2 and 4.0.0-dev.1 versions and relevant dependencies
(3.1.2 requires CMake 2.8.12). Updated all versions to use sha256 hashes.
Addressed issue specific to clang compiler on macOS.

### CODE REVIEW

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 1]
```diff
@@ -33,18 +33,29 @@ class Sundials(CMakePackage):
...
     # Versions
     # ==========================================================================
 
-    version('4.0.0-dev', '1d4b538721b84ebc91ce7ad92d94beae')
-    version('3.1.1', 'e63f4de0be5be97f750b30b0fa11ef34', preferred=True)
-    version('3.1.0', '1a84ca41c7f71067e03d519ddbcd9dae')
-    version('3.0.0', '5163a44cedd7398bddda442ba00313b8')
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
-    version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
+    version('4.0.0-dev.1',
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/sundials/package.py]: Note that you don't have to wrap these -- we add exceptions for directives like `version` and for lines that contain long hashes.

Comment (balos1) [reply var/spack/repos/builtin/packages/sundials/package.py]: Is it preferred to have them unwrapped? If so, I can update the PR.

**Comment by balos1** [comment]
The Travis build seems to be stuck 'pending'.

**Comment by scheibelp** [comment]
> The Travis build seems to be stuck 'pending'.

Thanks for the heads up - I restarted the build

**Pull request merged**